### PR TITLE
Add shared cookie utility

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

**Comment by designjockey** [comment]
Could you please comment on what was refactored and why?

File: util/cookie.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        if (currentCookie.substring(0, name.length + 1) === (name + '=')) {
+          cookieValue = decodeURIComponent(currentCookie.substring(name.length + 1));
+          break;
+        }
+      }
+    }
+    return cookieValue;
+  }
+
+  function getExpiration(option) {
```
Comment (kaicataldo) [inline util/cookie.js]: This was moved out of the main `cookie` function because it failed JSHint's cyclomatic complexity limit.

**Comment by designjockey** [comment]
:+1:

File: util/cookie.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   * Taken from jQuery cookie to do easier cookie manipulation without pulling down more files
+   *
+   * @deprecated use set or get instead
+   * @param {String} name
+   * @param {Mixed} value
+   * @param {Object} options
+   *
+   * @returns {String|null}
+   */
+  cookie = function(name, value, options) {
```
Comment (mikesherov) [inline util/cookie.js]: we should not include this form.

Comment (mikesherov) [inline util/cookie.js]: the export should be an object with `get` and `set` methods, no reason to keep the legacy function around.

File: test/helpers/cookie.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+define([
```
Comment (mikesherov) [inline test/helpers/cookie.js]: this helper doesn't make sense for BeFF. You should inline this into the cookie test suite.

File: util/cookie.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      }
+      else {
+        date = option;
+      }
+      expires = '; expires=' + date.toUTCString(); // use expires attribute, max-age is not supported by IE
+    }
+    return expires;
+  }
+
+  var cookie = {
```
Comment (mikesherov) [inline util/cookie.js]: no need for a var declaration here. Just return the object.

File: util/cookie.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    if (option && (typeof option === 'number' || option.toUTCString)) {
+      if (typeof option === 'number') {
+        date = new Date();
+        date.setTime(date.getTime() + (option * 24 * 60 * 60 * 1000));
+      }
+      else {
+        date = option;
+      }
+      expires = '; expires=' + date.toUTCString(); // use expires attribute, max-age is not supported by IE
```
Comment (mikesherov) [inline util/cookie.js]: `return '; expires=' + date.toUTCString();`

File: util/cookie.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    return expires;
```
Comment (mikesherov) [inline util/cookie.js]: `return '';`

File: util/cookie.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
+define([], function() {
+  'use strict';
+
+  /**
+   * Calculates cookie expiration value
+   * @param {Number|Date} option
+   * @returns {String}
+   */
+  function getExpiration(option) {
+    var expires = '';
```
Comment (mikesherov) [inline util/cookie.js]: get rid of this var declaration.

File: util/cookie.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+     */
+    get: function(name) {
+      var cookieValue = null;
+      if (name && document.cookie && document.cookie !== '') {
+        var cookies = document.cookie.split(';');
+        for (var i = 0; i < cookies.length; i++) {
+          var currentCookie = cookies[i].trim();
+          // Does this cookie string begin with the name we want?
+          if (currentCookie.substring(0, name.length + 1) === (name + '=')) {
+            cookieValue = decodeURIComponent(currentCookie.substring(name.length + 1));
```
Comment (mikesherov) [inline util/cookie.js]: `return decodeURIComponent(currentCookie.substring(name.length + 1));`

File: util/cookie.js (hunk 4 above)
Comment (kaicataldo) [inline util/cookie.js]: Thought it might make it clearer giving it a name, but I like your way better

File: util/cookie.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+            break;
+          }
+        }
+      }
+      return cookieValue;
```
Comment (mikesherov) [inline util/cookie.js]: `return null;`

File: util/cookie.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  var cookie = {
+    /**
+     * Gets cookie with given name
+     * @param {String} name
+     * @returns {String|null}
+     */
+    get: function(name) {
+      var cookieValue = null;
```
Comment (mikesherov) [inline util/cookie.js]: get rid of var with the other changes.

File: test/specs/util/cookie.js [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+    beforeEach(function() {
+      helper = CookieHelper.init();
+    });
+
+    afterEach(function() {
+      helper.destroy();
+    });
+
+    it('sets the value', function() {
```
Comment (mikesherov) [inline test/specs/util/cookie.js]: move all the existing it's into either the get or the set describe.

Comment (mikesherov) [inline test/specs/util/cookie.js]: this seems like a dupe test.

File: test/specs/util/cookie.js [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+      var value = 'mine=1';
+      expect(document.cookie).not.toContain(value);
+      cookie.set('mine', 1);
+      expect(document.cookie).toContain(value);
+    });
+
+    it('gets the value', function() {
```
Comment (mikesherov) [inline test/specs/util/cookie.js]: looks like a dupe test,

File: test/specs/util/cookie.js [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+      cookie.set('mine', 2);
+      expect(cookie.get('mine')).toEqual('2');
+    });
+
+    it('returns null if undefined is requested', function() {
```
Comment (mikesherov) [inline test/specs/util/cookie.js]: move to the `#get` describe

File: test/specs/util/cookie.js [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+      expect(cookie.get()).toBeNull();
+    });
+
+    it('removes the value', function(done) {
```
Comment (mikesherov) [inline test/specs/util/cookie.js]: move to the `set` describe.

File: test/specs/util/cookie.js (hunk 12 above)
Comment (kaicataldo) [inline test/specs/util/cookie.js]: Oh weird, I missed that when I was copying it over. Good catch.

File: util/cookie.js [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * Gets cookie with given name
+     * @param {String} name
+     * @returns {String|null}
+     */
+    get: function(name) {
+      if (name && document.cookie && document.cookie !== '') {
+        var cookies = document.cookie.split(';');
+        for (var i = 0; i < cookies.length; i++) {
+          var currentCookie = cookies[i].trim();
```
Comment (mikesherov) [inline util/cookie.js]: move this var up to onevar with `cookies`

File: util/cookie.js [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+  return {
+    /**
+     * Gets cookie with given name
+     * @param {String} name
+     * @returns {String|null}
+     */
+    get: function(name) {
+      if (name && document.cookie && document.cookie !== '') {
+        var cookies = document.cookie.split(';');
+        for (var i = 0; i < cookies.length; i++) {
```
Comment (mikesherov) [inline util/cookie.js]: move `var i` up to one var with `cookies`

File: test/specs/util/cookie.js [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+  'use strict';
+
+  /**
+   * Ensures there are no cookies to interfere with tests
+   */
+  function resetCookies() {
+    var cookies = document.cookie.split(';');
+
+    cookies.forEach(function(cookie) {
+      cookie = $.trim(cookie);
```
Comment (mikesherov) [inline test/specs/util/cookie.js]: cookie = cookie.trim();

File: test/specs/util/cookie.js [hunk 18]
```diff
@@ -0,0 +1,61 @@
+define([
+  'jquery',
```
Comment (mikesherov) [inline test/specs/util/cookie.js]: drop the jquery dep.

File: test/specs/util/cookie.js [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+  'util/cookie'
+], function($, cookie) {
```
Comment (mikesherov) [inline test/specs/util/cookie.js]: drop the jquery dep.

**Comment by mikesherov** [comment]
:+1:

**Pull request merged**